src/mmc_windows.c

```diff
@@ -78,63 +41,56 @@ static int mmc_query_volume_path(const WCHAR *volume_path, struct mmc_device *de
  */
 int mmc_scan_for_devices(struct mmc_device *devices, int max_devices)
 {
-    DWORD  Error                = ERROR_SUCCESS;
-    HANDLE FindHandle           = INVALID_HANDLE_VALUE;
-    size_t Index                = 0;
-    BOOL   Success              = FALSE;
-    WCHAR  VolumeName[MAX_PATH] = L"";
+    WCHAR drive_path[MAX_PATH] = L"";
+    HANDLE drive_handle;
+    DWORD bytes_returned;
+    DISK_GEOMETRY_EX geometry;
+    BOOL status;
+    struct mmc_device *device;
+
+    // There's not an API to request all the PhysicalDrives, but they are
+    // sequentially enumerated and become invalid if they're removed.
+    // So we just walk the list until we see 5 missing drives in a row,
+    // assuming there won't be any more past that point.
     int device_count = 0;
+    int consecutive_invalid_drives = 0;
+    for(int i=0; consecutive_invalid_drives < 5 && device_count < max_devices ; i++) {
```

Comment:
Fix the spacing here: `for (int i = 0; consecutive_invalid_drives < 5 && device_count < max_devices; i++) {`